### ISSUE TITLE
NullJsonWebToken throw NPE

### ISSUE THREAD

**Issue by jie-huang** [smallrye/smallrye-jwt#585]
NullJsonWebToken implements JsonWebToken.

But, getExpirationTime() and getIssuedAtTime() in JsonWebToken both return primitive `long` type.
So, the implementation of <T> T getClaim(String claimName) should not return null in such cases.
Otherwise, it triggers NPE when unboxing it.

Error message is like
---------------------------------
Cannot invoke "java.lang.Long.longValue()" because the return value of "org.eclipse.microprofile.jwt.JsonWebToken.getClaim(org.eclipse.microprofile.jwt.Claims)" is null

**Comment by sberyozkin** [comment]
But it is probably correct. Accessing a null token should lead to NPE, `NullJsonWebToken` only exists because a request scoped `null` token has to be represented somehow

**Comment by jie-huang** [comment]
Not really. NullJsonWebToken is a null token but it is not a Java null.
Since NullJsonWebToken implements JsonWebToken, when it is not a Java null, it should have consistent behavior for different methods.
e.g. getRawToken() returns null. It is reasonable because it does not have a token.
But, getExpirationTime() throws NPE. Personally, such cases return -1 or 0 could be acceptable and easy for people to use it.

The inconsistency causes problems some time. e.g. When using jackson to convert a non-null JsonWebToken to json, it triggers NPE even config is to include "non-null".

**Comment by sberyozkin** [comment]
@jie-huang But `0` or `-1` is not an indicator that the token does not exists. The raw token is null, so how is returning `0` or `-1` can be considered consistent ?

What I'd also like to check, what is the actual use case ? Are you working with Quarkus ? Is it on the public path, does the endpoint method require the authentication and/or authorization ? If it is a protected endpoint then an injected token would never be null.
And if it is a public endpoint then it is really a responsibility of the endpoint itself to ensure the consumed token is non null

**Comment by jie-huang** [comment]
I found this NPE when I work on some tools with Quarkus.
I tried to return a JsonObject of the JsonWebToken (with/out authentication).
Like what you said, when it is not authenticated, it is a NullJsonWebToken object.
I tried to customize the conversion of NullJsonWebToken object. Unfortunately, it did not work because both NullJsonWebToken and JsonWebToken go to the same _serialize()_ method. And, _instanceof_ does not work.
The JsonWebToken object passed to _serialize()_ method has some strange class like _org.eclipse.microprofile.jwt.OidcJsonWebTokenProducer_ProducerMethod_currentAccessToken_6cc9acf87bb9cb50add3593d880ba325b4a9f587_ClientProxy_.

Now, I understand why it happens and work-around it.
It is just not so natural since it cannot work with normal Json Serialization.
Thanks
